lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js

```diff
@@ -0,0 +1,253 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-len, max-params */
+
+'use strict';
+
+// MODULES //
+
+var idamax = require( '@stdlib/blas/base/idamax' ).ndarray;
+var dasum = require( '@stdlib/blas/base/dasum' ).ndarray;
+var dcopy = require( '@stdlib/blas/base/dcopy' ).ndarray;
+var floor = require( '@stdlib/math/base/special/floor' );
+var abs = require( '@stdlib/math/base/special/abs' );
+
+
+// FUNCTIONS //
+
+/**
+* Truncates a number to whole number.
+*
+* @private
+* @param {number} x - number to truncate
+* @returns {number} truncated number
+*
+* @example
+* var v = aint( 3.14 );
+* // returns 3.0
+*/
+function aint( x ) {
+	var sign;
+	var ax;
+
+	ax = abs( x );
+	if ( x >= 0.0 ) {
+		sign = 1;
+	} else {
+		sign = -1;
+	}
+
+	if ( ax < 1.0 ) {
+		return 0;
+	}
+	if ( ax >= 1.0 ) {
+		return sign * floor( ax );
+	}
+}
+
+/**
+* Rounds a number to the nearest whole number.
+*
+* @private
+* @param {number} x - number to round
+* @returns {number} rounded number
+*
+* @example
+* var v = nint( 3.14 );
+* // returns 3.0
+*
+* @example
+* var v = nint( 9.99999 );
+* // returns 10.0
+*/
+function nint( x ) {
```

Comment:
Isn't this just `@stdlib/math/base/special/round`?

Reply:
As per https://github.com/stdlib-js/stdlib/pull/745#issuecomment-1994352608, round returns decimal towards `0.0` , here is a small example:

```console
% cat a.f90 
program main
        print *, nint(1.5)
        print *, nint(-1.5)
end program
% gfortran a.f90&& ./a.out       
           2
          -2
% cat a.js                
var round = require('@stdlib/math/base/special/round');
console.log( round(1.5) );
console.log( round(-1.5) );
% node a.js               
2
-1
```

Reply:
With the current implementation:

```js
var floor = require( '@stdlib/math/base/special/floor' );
var abs = require( '@stdlib/math/base/special/abs' );

function aint( x ) {
	var sign;
	var ax;

	ax = abs( x );
	if ( x >= 0.0 ) {
		sign = 1;
	} else {
		sign = -1;
	}

	if ( ax < 1.0 ) {
		return 0;
	}
	if ( ax >= 1.0 ) {
		return sign * floor( ax );
	}
}

function nint( x ) {
	if ( x >= 0.0 ) {
		return aint( x + 0.5 );
	}
	return aint( x - 0.5 );
}

var round = require('@stdlib/math/base/special/round');
console.log( round(1.5) );
console.log( round(-1.5) );

console.log( "NINT" );
console.log( nint(1.5) );
console.log( nint(-1.5) );
```

We get:

```console
% node a.js               
2
-1
NINT
2
-2

```

Reply:
Ah, right. Rounding behavior differs between langs and platforms. We should add base special math functions for the different rounding behaviors. This isn't the first time this has arisen, and such functions are included in IEEE 754, so they'd be good to have.

Reply:
Reference: https://gcc.gnu.org/onlinedocs/gfortran/AINT.html, https://gcc.gnu.org/onlinedocs//gfortran/NINT.html

Reply:
In short, fine to keep these as is, but we should also create todo issues to add the specialized rounding functions, and add TODOs here to indicate that `nint` and `aint` should be replaced by future stdlib utilities.